subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCachePublishingIntegrationTest.groovy

```diff
@@ -16,14 +16,22 @@
 
 package org.gradle.configurationcache
 
+import org.gradle.internal.credentials.DefaultPasswordCredentials
+import org.gradle.test.fixtures.server.http.HttpServer
+import org.gradle.test.fixtures.server.http.MavenHttpRepository
+import org.gradle.util.internal.GUtil
+import org.junit.Rule
 
 import static org.gradle.util.internal.GFileUtils.deleteDirectory
 import static org.gradle.util.internal.GFileUtils.listFiles
 
 class ConfigurationCachePublishingIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
 
+    @Rule
+    public final HttpServer server = new HttpServer()
+
     def setup() {
-        buildFile << """
+        buildFile.setText """
```

Comment:
Why?

Reply:
Idempotency/predictability of setText vs append

Reply:
I'd actually favor the more modern version,`buildFile "..."` over `buildFile << "..."`.

Reply:
I noticed that, but am unclear on whether the common intent is to append or to set the contents. Because /if/ it is the latter, setText() seems the most appropriate. But instead of going around changing the use of `<<` with `setText`, one simple change would be to change `buildFile(String)` to call `setText` instead of `<<`, assuming that does not break any tests. Note that `buildFile(String)` and `settingsFile(String)` both use `<<`, whereas `buildScript(String)` uses `setText`. I am wondering whether it is just accidental lack of uniformity vs intentional.  Hoping for the former, a  test run should show.

Reply:
I understand the overall preference for `<<` as it allows different concerns (`dependencies`, `repositories`, etc) to be handled at different levels of the test hierarchy. Of course that's not always desirable but I don't have strong feelings either way. 

If you feel strongly about it I suggest proposing the change in a separate PR.